arctic/store/version_store.py

```diff
@@ -524,9 +568,123 @@ def _publish_change(self, symbol, version):
+
+        self._publish_change(symbol, version)
+
+        return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=0, data=None, metadata=metadata)
+
+    @mongo_retry
+    def _update_metadata_history(self, symbol, metadata, metadata_policy=lambda old, new: new, **kwargs):
+        """
+        Update .metadata entry for `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        metadata_policy :
+            function(old, new):
+                return final metadata to write
+            Default: overwrite
+        kwargs :
+            passed through to the write handler
+
+        Returns
+        -------
+        `metadata` if metadata is written, None if not.
+        """
+        if not metadata:
+            return None
+        elif self.has_symbol(symbol, metadata=True):
+            old_metadata = self.read_metadata(symbol).metadata
+            new_metadata = metadata_policy(old_metadata, metadata)
+            if not new_metadata or old_metadata == new_metadata:
+                logger.info('No change to metadata.')
+                return None
+            else:
+                metadata = new_metadata
+        now = dt.now()
+        self._metadata.find_one_and_update({'symbol': symbol}, {'$set': {'_end_time': now}},
+                                            sort=[('_start_time', pymongo.DESCENDING)])
+        self._write_metadata_history(symbol, metadata, now, **kwargs)
+        return metadata
+
+    @mongo_retry
+    def update_metadata(self, symbol, metadata, metadata_policy=lambda old, new: new, **kwargs):
+        """
+        Update .metadata entry for `symbol` without changing the data
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        metadata_policy :
+            function(old, new):
+                return final metadata to write
+            Default: overwrite
+        kwargs :
+            passed through to the write handler
+
+        Returns
+        -------
+        VersionedItem containing .metadata
+        """
+        if not metadata:
+            raise ValueError('Metadata not specified.')
+        new_metadata = self._update_metadata_history(symbol, metadata, metadata_policy, **kwargs)
+        if new_metadata:
+            data = self.read(symbol).data
+            return self.write_data(symbol, data, new_metadata, **kwargs)
+        logger.warning('No changes detected. Returning current entry.')
+        old = self.read_metadata(symbol)
+        return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(),
+                                 version=old.version, metadata=old.metadata, data=None)
+
+    @mongo_retry
+    def write_data(self, symbol, data, metadata=None, prune_previous_version=True, **kwargs):
```

Comment:
``write_data`` used to modify documents in .versions without making changes to .metadata where ``write`` also updates .metadata. This is so that legacy code using ``write`` will also update .metadata.